src/projections/isea.cpp

```diff
@@ -1105,5 +1108,456 @@ PJ *PJ_PROJECTION(isea) {
 #undef F_RAD
 #undef TABLE_G
 #undef TABLE_H
+
+/*
+   Icosahedron Snyder equal-area (ISEA) Inverse Projection
+   --------------------------------------------------------------------------
+   This inverse projection was adapted from Java and eC by
+   Jérôme Jacovella-St-Louis, originally from the Franz-Benjamin Mocnik's ISEA
+   implementation found at
+   https://github.com/mocnik-science/geogrid/blob/master/
+    src/main/java/org/giscience/utils/geogrid/projections/ISEAProjection.java
+   with the following license:
+
+   MIT License
+
+   Copyright (c) 2017-2019 Heidelberg University
+
+   Permission is hereby granted, free of charge, to any person obtaining a copy
+   of this software and associated documentation files (the "Software"), to deal
+   in the Software without restriction, including without limitation the rights
+   to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+   copies of the Software, and to permit persons to whom the Software is
+   furnished to do so, subject to the following conditions:
+
+   The above copyright notice and this permission notice shall be included in
+   all copies or substantial portions of the Software.
+
+   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+   SOFTWARE.
+*/
+
+/* The ISEA projection a projects a sphere on the icosahedron. Thereby the size
+ * of areas mapped to the icosahedron are preserved. Angles and distances are
+ * however slightly distorted. The angular distortion is below 17.27 degree, and
+ * the scale variation is less than 16.3 per cent.
+ *
+ * The projection has been proposed and has been described in detail by:
+ *
+ * John P. Snyder: An equal-area map projection for polyhedral globes.
+ * Cartographica, 29(1), 10–21, 1992. doi:10.3138/27H7-8K88-4882-1752
+ *
+ * Another description and improvements can be found in:
+ *
+ * Erika Harrison, Ali Mahdavi-Amiri, and Faramarz Samavati: Optimization of
+ * inverse Snyder polyhedral projection. International Conference on Cyberworlds
+ * 2011. doi:10.1109/CW.2011.36
+ *
+ * Erika Harrison, Ali Mahdavi-Amiri, and Faramarz Samavati: Analysis of inverse
+ * Snyder optimizations. In: Marina L. Gavrilova, and C. J. Kenneth Tan (Eds):
+ * Transactions on Computational Science XVI. Heidelberg, Springer, 2012. pp.
+ * 134–148. doi:10.1007/978-3-642-32663-9_8
+ */
+
+struct GeoPoint {
+    double lat, lon;
+}; // In radians
+
+#define Pi 3.1415926535897932384626433832795028841971
+
+#define Degrees(x) ((x)*Pi / 180)
+
+#define wgs84InvFlattening 298.257223563
```

Comment:
All of those WGS84 related parameters should likely be replaced by using P->a, P->b, P->es parameters

Reply:
That is turning out into quite an overhaul, but it will handle arbitrary `+R=` values... Working on it.

Reply:
I did that, but that turned out into trying to improve round-trip precision (including for the previous test cases), and as a result I had to merge some the constants with the forward projection.

Now should I try to implement this automatic geodetic ==> authalic when the a non-spherical ellipsoid is in use?
Do we keep that for a separate commit / PR?

Regarding authalic ==> geodetic conversion, there's no good reason that `pj_authset()` is doing a dynamic memory allocation for 3 doubles -- those `double *apa;` should just be `double apa[3]` in _cea.cpp_, _eqearth.cpp_, _healpix.cpp_ and _laea.cpp_ and passed as a return pointer to `pj_authset()`. Can I fix that? 

`cea_e_forward()` / `cea_e_inverse()` maybe the simplest example of this automatic geodetic / authalic conversion.

Reply:
> Now should I try to implement this automatic geodetic ==> authalic when a non-spherical ellipsoid is in use?
Do we keep that for a separate commit / PR?

Yes perhaps keep that for a follow-up PR

> Regarding authalic ==> geodetic conversion, there's no good reason that `pj_authset()` is doing a dynamic memory allocation for 3 doubles -- those `double *apa;` should just be `double apa[3]` in _cea.cpp_, _eqearth.cpp_, _healpix.cpp_ and _laea.cpp_ and passed as a return pointer to `pj_authset()`. Can I fix that?

That's more a micro-optimization that will likely not measurable. Should be kept for a later PR if you want to tackle that